src/main/capi/prepared-c.cpp

```diff
@@ -140,6 +140,25 @@ duckdb_type duckdb_param_type(duckdb_prepared_statement prepared_statement, idx_
 	return DUCKDB_TYPE_INVALID;
 }
 
+duckdb_logical_type duckdb_param_logical_type(duckdb_prepared_statement prepared_statement, idx_t param_idx) {
```

Comment:
Thanks, can we replace the body of `duckdb_param_type` with this? Seeing as this is currently duplicating a lot of that logic

Inside `duckdb_param_type` we would only need to call `ConvertCPPTypeToC` on the result and then clean up the created `duckdb_logical_type`.